pkg/apis/resource/validation/validation.go

```diff
@@ -46,6 +46,8 @@ var (
 	validateDeviceName      = corevalidation.ValidateDNS1123Label
 	validateDeviceClassName = corevalidation.ValidateDNS1123Subdomain
 	validateRequestName     = corevalidation.ValidateDNS1123Label
+
+	attributeAndCapacityMaxKeyLength = resource.DeviceMaxDomainLength + 1 + resource.DeviceMaxIDLength
```

Comment:
It might be worth adding a comment here that this is for `domain/ID`

Reply:
Thanks for the review, added comment